### ISSUE TITLE
Holding button during beep loop causes crash

### ISSUE THREAD

**Issue by ScttB** [robhagemans/pcbasic#46]
![image](https://user-images.githubusercontent.com/6832106/32034308-c555b944-b9c6-11e7-81ab-9ddbca188afd.png)
Specifically: My code will let you hold down a button and it will beep after a counter goes up to 22 (starting at 3), so holding down the button when the counter is up will cause this crash.
I made a special program that's only purpose is to cause this crash:
```
10 LET KEE$=INKEY$
20 IF KEE$="c" THEN: BEEP
30 GOTO 10
```
Hold C to trigger it.
I have tested this in GWBASIC, and it does not cause any crash, only extreme ear pain.

**Comment by robhagemans** [comment]
Hi, thanks for the report and the trigger code, I'll have a look

**Comment by robhagemans** [comment]
Does not occur in 16.12, but does occur in 15.08.12.

The problem appears to be something related to the keyboard being checked in the wait loop after the `BEEP` statement:

```
  File "/home/rob/Projects/basic-project/branch1508/pcbasic/sound.py", line 169, in play_sound
    self.wait_music(15)
  File "/home/rob/Projects/basic-project/branch1508/pcbasic/sound.py", line 176, in wait_music
    backend.wait()
  File "/home/rob/Projects/basic-project/branch1508/pcbasic/backend.py", line 61, in wait
    check_events()
  File "/home/rob/Projects/basic-project/branch1508/pcbasic/backend.py", line 70, in check_events
    video.check_events()
  File "/home/rob/Projects/basic-project/branch1508/pcbasic/video_pygame.py", line 926, in check_events
    handle_key_down(event)
  File "/home/rob/Projects/basic-project/branch1508/pcbasic/video_pygame.py", line 1064, in handle_key_down
    backend.key_down(scan, c)
  File "/home/rob/Projects/basic-project/branch1508/pcbasic/backend.py", line 607, in key_down
    state.console_state.keyb.key_down(scan, eascii, check_full)
  File "/home/rob/Projects/basic-project/branch1508/pcbasic/backend.py", line 580, in key_down
    self.insert_chars(eascii, check_full=True)
  File "/home/rob/Projects/basic-project/branch1508/pcbasic/backend.py", line 497, in insert_chars
    state.console_state.sound.play_sound(800, 0.01)
  File "/home/rob/Projects/basic-project/branch1508/pcbasic/sound.py", line 169, in play_sound
    self.wait_music(15)
  File "/home/rob/Projects/basic-project/branch1508/pcbasic/sound.py", line 176, in wait_music
    backend.wait()
  File "/home/rob/Projects/basic-project/branch1508/pcbasic/backend.py", line 61, in wait
    check_events()
  File "/home/rob/Projects/basic-project/branch1508/pcbasic/backend.py", line 70, in check_events
    video.check_events()
  File "/home/rob/Projects/basic-project/branch1508/pcbasic/video_pygame.py", line 926, in check_events
    handle_key_down(event)
  File "/home/rob/Projects/basic-project/branch1508/pcbasic/video_pygame.py", line 1064, in handle_key_down
    backend.key_down(scan, c)
  File "/home/rob/Projects/basic-project/branch1508/pcbasic/backend.py", line 607, in key_down
    state.console_state.keyb.key_down(scan, eascii, check_full)
  File "/home/rob/Projects/basic-project/branch1508/pcbasic/backend.py", line 580, in key_down
    self.insert_chars(eascii, check_full=True)
  File "/home/rob/Projects/basic-project/branch1508/pcbasic/backend.py", line 497, in insert_chars
    state.console_state.sound.play_sound(800, 0.01)
  File "/home/rob/Projects/basic-project/branch1508/pcbasic/sound.py", line 161, in play_sound
    tone = AudioEvent(AUDIO_TONE, (frequency, duration, fill, loop, volume))
RuntimeError: maximum recursion depth exceeded
```

**Comment by robhagemans** [comment]
Fixed by 598537a
